tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoTargetMapping.java

```diff
@@ -37,13 +37,18 @@ protected String getPackaging() {
         return PACKAGING;
     }
 
+    @Override
+    public float getPriority() {
+        return 20;
+    }
+
     @Override
     protected boolean isValidLocation(String location) {
         return location.endsWith(TARGET_EXTENSION);
     }
 
     @Override
-    protected File getPrimaryArtifact(File dir) {
+    public File locatePom(File dir) {
```

Comment:
Is there any reason here (and on other places) to override locatePom instead of using getPrimaryArtifact?

Reply:
This was because with the higher Prioritized PomXMLMapping the check for a pom.xml was not necessary in the `locatePom()` implementation in the super-class which was just calling `getPrimaryArtifact()`, so `locatePom()` could have been called directly. But since we cannot introduce a PomXMLMapping, that has to stay as it is and I reverted it.